dpdk: retry to download source on transient connectivity errors

ader1990 opened the pull request:
Fixes issue https://github.com/LIS/LISAv2/issues/286

Sometimes, there are network connectivity or server down issues when downloading DPDK sources. A retry should be added in those scenarios.

Example:

```
wget http://git.dpdk.org/dpdk-stable/snapshot/dpdk-stable-18.11.2-rc1.tar.xz
Resolving git.dpdk.org (git.dpdk.org)... 92.243.14.124
Connecting to git.dpdk.org (git.dpdk.org)|92.243.14.124|:80... connected.
HTTP request sent, awaiting response... 504 Gateway Time-out
2019-05-30 19:49:33 ERROR 504: Gateway Time-out.
```

lizzha commented:
Looks OK, do you have the test log when an exception is triggered?

juhlee-microsoft commented:
> Looks OK, do you have the test log when an exception is triggered?

No, can you make a suggestion? I don't have any top of my head

ader1990 committed:
dpdk: retry to download source on transient connectivity errors

If the wget call fails, a retry is needed in case the server
does not support multiple calls or if the server is
not responding correctly.
